var/spack/repos/builtin/packages/fsl/package.py

```diff
@@ -8,49 +8,163 @@
+            build_settings.filter(r'(^CUDA_INSTALLATION)\s*=.*',
+                                  r'\1 = {0}'.format(cuda_installation))
+            build_settings.filter(r'(^LIB_CUDA)\s*=.*',
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation,
+                                          'lib64')))
+            build_settings.filter(r'(^INC_CUDA)\s*=.*',
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation,
+                                          'include')))
+            build_settings.filter(r'(^NVCC11)\s*=.*',
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation, 'bin',
+                                          'nvcc')))
+            build_settings.filter(r'(^NVCC)\s*=.*',
+                                  r'\1 = {0}'.format(
+                                      join_path(
+                                          cuda_installation, 'bin',
+                                          'nvcc')))
+            build_settings.filter(r'(^GENCODE_FLAGS)\s*=.*',
+                                  r'\1 = {0}'.format(cuda_gencode))
+
+            if self.spec.satisfies('@6:'):
+                build_settings.filter(r'(^EDDYBUILDPARAMETERS)\s*=.*',
+                                      r'\1 = "cuda=1" "cpu=1"')
+                build_settings.filter(r'(^fdt_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=1')
+                build_settings.filter(r'(^ptx2_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=1')
+            else:
+                with open(settings_file, 'a') as f:
+                    f.write('COMPILE_GPU=1\n')
+        else:
+            build_settings.filter(r'^CUDA_INSTALLATION', '#CUDA_INSTALLATION')
+            build_settings.filter(r'^GENCODE_FLAGS', '#GENCODE_FLAGS')
+            build_settings.filter(r'^LIB_CUDA', '#LIB_CUDA')
+            build_settings.filter(r'^INC_CUDA', '#INC_CUDA')
+            build_settings.filter(r'^NVCC', '#NVCC')
+
+            if self.spec.satisfies('@6:'):
+                build_settings.filter(r'(^EDDYBUILDPARAMETERS)\s*=.*',
+                                      r'\1 = "cpu=1"')
+                build_settings.filter(r'(^fdt_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=0')
+                build_settings.filter(r'(^ptx2_MASTERBUILD)\s*=.*',
+                                      r'\1 = COMPILE_GPU=0')
 
     def install(self, spec, prefix):
         build = Executable('./build')
         build()
 
         install_tree('.', prefix)
 
+    @run_after('install')
+    def postinstall(self):
+        del os.environ['PYTHONPATH']
+        del os.environ['PYTHONHOME']
```

Comment:
Can you put this within the:
```python
with spack.util.environment.preserve_environment('PYTHONPATH', 'PYTHONHOME'):
    ....
```
context manager? Asking to ensure that the environment is restores after executing the script.

Reply:
Okay; I am not really familiar with that. The only other package that I see with that is octave so I assume that is a good example.